tests/unit/multi-currency/compatibility/test-class-woocommerce-subscriptions.php

```diff
@@ -286,13 +286,20 @@ public function test_override_selected_currency_return_currency_code_for_switch_
 		$this->mock_wcs_get_order_type_cart_items( false );
 
 		// Set up an order with a non-default currency.
-		// This might need to be a subscription object, not an order.
-		$order = WC_Helper_Order::create_order();
-		$order->set_currency( 'JPY' );
-		$order->save();
+		// Note we're using a WC_Order as a stand-in for a true WC_Subscription.
+		$mock_subscription = WC_Helper_Order::create_order();
+		$mock_subscription->set_currency( 'JPY' );
+		$mock_subscription->save();
+
+		// Mock wcs_get_subscription to return our mock subscription.
+		WC_Subscriptions::set_wcs_get_subscription(
```

Comment:
Using a mock `WC_Subscription` rather than `WC_Order`, as the `override_selected_currency` expects. This is consistent with neighbouring tests.